phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/MultiThreadedRunner.java

```diff
@@ -148,34 +161,55 @@ private void timedQuery() throws Exception {
             boolean isQuery = statement.execute();
             if (isQuery) {
                 rs = statement.getResultSet();
-                while (rs.next()) {
-                    if (null != query.getExpectedAggregateRowCount()) {
-                        if (rs.getLong(1) != query.getExpectedAggregateRowCount())
-                            throw new RuntimeException(
-                                    "Aggregate count " + rs.getLong(1) + " does not match expected "
-                                            + query.getExpectedAggregateRowCount());
-                    }
-
-                    if (isSelectCountStatement) {
-                        resultRowCount = rs.getLong(1);
-                    } else {
-                        resultRowCount++;
-                    }
-                }
+                Pair<Long, Boolean> r = getResults(rs, queryIteration, isSelectCountStatement);
```

Comment:
As discussed, let's make this method return a Pair<Long, Long> where the second Long is an `endTime` which we then use to write to the csv later. That way the actual time we check and the time we report is the same. Also saves us unnecessary calls to `EnvironmentEdgeManager.currentTimeMs()`